core/state/snapshot/journal.go

```diff
@@ -33,9 +33,10 @@ import (
 	"github.com/ethereum/go-ethereum/triedb"
 )
 
-// 0: initial version
-// 1: destruct flag in diff layer is removed
-const journalVersion uint64 = 1
+const (
+	journalInitVersion    uint64 = 0 // initial version
+	journalCurrentVersion uint64 = 1 // current version, with destruct flag (in diff layers) removed
```

Comment:
Wouldn't it be better to use e.g. 
```suggestion
	journalV0    uint64 = 0 // initial version
	journalV1    uint64 = 1 // current version, with destruct flag (in diff layers) removed
	
	journalCurrentVersion = journalV1
```
Then those parts of the code where we _want_ the code to remain unchanged (follow latest) we use `journalCurrentVersion`, and whenever we want to write code for specific version, we use those.

Reply:
sgtm, i will apply